### PR TITLE
User proper JSON for /api/v1/data

### PR DESCRIPTION
**Pull request by simonnagl**
This fixes issue #83
The output with options=google_json does not change.

@ktsaou Sorry for delaying documentation but documentation needs a special type of mood ;) Please be patient... 

I am pretty sure this is correct because printing new Date(... is very Google specific and we should not use that in any other output.

Of course as mentioned in #83 we might need tests to ensure our JSON API does not change. I want to start a discussion here what might be the best solution implementing such tests... Any ideas?

### CODE REVIEW

**Comment by ktsaou** [comment]
thanks! does the `http://ip.of.netdata:19999/old/` (old dashboard) work?

**Comment by simonnagl** [comment]
Wow! I did not even no something like this exists... 

http://ip.of.netdata:19999/old/index.html works. http://ip.of.netdata:19999/old/ prints errors not finding index.html etc. ...

Charts are showing the same data as the new dashboard.

**Comment by ktsaou** [comment]
> Wow! I did not even no something like this exists...

This was netdata 14 months ago... google charts are reaaaaaaaaaally slow...

**Comment by ktsaou** [comment]
> http://ip.of.netdata:19999/old/ prints errors not finding index.html

strange... works for me...

**Comment by ktsaou** [comment]
> Charts are showing the same data as the new dashboard.

yes

**Comment by simonnagl** [comment]
http://ip.of.netdata:19999/old/ works
http://ip.of.netdata:19999/old works not.

Which might be obvious.

**Comment by ktsaou** [comment]
> I want to start a discussion here what might be the best solution implementing such tests... Any ideas?

Ideas:

1. We could have a special module in netdata that generates a pre-agreed series of data. Then we could use the api to get its output and compare it with mockups we know are good.

2. We could add another `-W` option that fills a few charts with predefined data and then have another `-W` option to emulate API calls that return API responses. This mode could be used in travis to verify the code is right.

   I have also registered netdata to https://github.com/ktsaou/oss-fuzz which could be used to feed netdata (i.e. API requests) with random garbage to verify it does not crash. But I haven't found the time yet to do anything with it.

I you are interested in this area, it would be perfect to do something about it. Testing netdata like that will allow us to work faster (currently, I am manually running a large set of tests every time I change something).

**Comment by ktsaou** [comment]
let me know when you want me to merge this.

**Comment by ktsaou** [comment]
(btw, please remove my username from this commit: https://github.com/simonnagl/netdata/commit/0e8fe6bff86647c9e8af0aa3e8de350db8b0d14a - I keep getting emails from github about it).

**Comment by simonnagl** [comment]
> btw, please remove my username from this commit: simonnagl@0e8fe6b - I keep getting emails from github about it

Please tell me if I fixed it.

> let me know when you want me to merge this.

If you are OK with the change please do so.

> We could have a special module in netdata that generates a pre-agreed series of data. Then we could use the api to get its output and compare it with mockups we know are good.

I agree.

> We could add another -W option that fills a few charts with predefined data and then have another -W option to emulate API calls that return API responses. This mode could be used in travis to verify the code is right.

I do not agree. I do not like building tests into the program itself. It makes it bigger more complex. I think we should we should develop another program for doing tests.
I did not found a good module for doing tests in C yet, so maybe we need to develop something ourself. For testing public API it might be better using another programming language which makes our live easier parsing for example JSON. I will do some research and open a new issue when finished.

As always, please be patient with my work ;)

**Comment by ktsaou** [comment]
> do not like building tests into the program itself. It makes it bigger more complex. I think we should we should develop another program for doing tests.

You are right, so let me rephrase it:

We could add another -W option that reads data from stdin to put into the database and another -W option that reads API URLs and prints the output on stdout. So, we could have external tests without loading the daemon (i.e. via travis).

> As always, please be patient with my work ;)

I am. Take your time.

**Comment by ktsaou** [comment]
> Please tell me if I fixed it.

I think you did. No more emails about it. Thank you!

**Comment by ktsaou** [comment]
merged! thanks!
